README.md

```diff
@@ -63,6 +68,95 @@ jobs:
           password: '${{ secrets.DbUserPassword }}'
 ```
 
+**Using Azure Service Principal authentication**
+```yml
+jobs:
+  migrate-database:
+    runs-on: [self-hosted, windows-2019]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Flyway
+        uses: actions/setup-flyway@v1
+        with:
+          version: 5.1.4
+
+      - name: Run Flyway Migrations
+        uses: im-open/run-flyway-command@v1.4.0
+        with:
+          db-server-name: 'localhost'
+          db-server-port: '1433'
+          db-name: 'LocalDb'
+          migration-files-path: './src/Database/Migrations'
+          flyway-command: 'migrate'
+          migration-history-table: 'dbo.MigrationHistory'
+          baseline-version: '0'
+          managed-schemas: 'dbo,MyCustomSchema'
+          enable-out-of-order: 'true'
+          use-azure-service-principal: 'true'
+          username: '${{ secrets.AZ_SERVICE_PRINCIPAL_ID }}'
+          password: '${{ secrets.AZ_SERVICE_PRINCIPAL_SECRET }}'
+```
+
+**Using Azure User-Assigned Managed Identity authentication**
+```yml
+jobs:
+  migrate-database:
+    runs-on: [self-hosted, windows-2019]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Flyway
+        uses: actions/setup-flyway@v1
+        with:
+          version: 5.1.4
+
+      - name: Run Flyway Migrations
+        uses: im-open/run-flyway-command@v1.4.0
+        with:
+          db-server-name: 'localhost'
+          db-server-port: '1433'
+          db-name: 'LocalDb'
+          migration-files-path: './src/Database/Migrations'
+          flyway-command: 'migrate'
+          migration-history-table: 'dbo.MigrationHistory'
+          baseline-version: '0'
+          managed-schemas: 'dbo,MyCustomSchema'
+          enable-out-of-order: 'true'
+          use-azure-managed-identity: 'true'
+          azure-msi-client-id: '${{ secrets.AZ_MSI_CLIENT_ID }}'
+```
+
+**Using Azure System-Assigned Managed Identity authentication**
+```yml
+jobs:
+  migrate-database:
+    runs-on: [self-hosted, windows-2019]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Flyway
+        uses: actions/setup-flyway@v1
+        with:
+          version: 5.1.4
+
+      - name: Run Flyway Migrations
+        uses: im-open/run-flyway-command@v1.4.0
+        with:
+          db-server-name: 'localhost'
+          db-server-port: '1433'
+          db-name: 'LocalDb'
+          migration-files-path: './src/Database/Migrations'
+          flyway-command: 'migrate'
+          migration-history-table: 'dbo.MigrationHistory'
+          baseline-version: '0'
+          managed-schemas: 'dbo,MyCustomSchema'
+          enable-out-of-order: 'true'
+          use-azure-managed-identity: 'true'
```

Comment:
This is probably a dumb question but how does it know what msi to use?  It doesn't seem like anything here would set that up.

Reply:
Good point. I imagine I need to do an `az login`. Truth be told I didn't test system-assigned identities, I just know it's possible from [this documentation](https://docs.microsoft.com/en-us/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver16#properties) (see the `authentication` prop).